classes/order/Order.php

```diff
@@ -1852,7 +1852,7 @@ public function addOrderPayment($amount_paid, $payment_method = null, $payment_t
         if ($order_payment->id_currency == $this->id_currency) {
             $this->total_paid_real += $order_payment->amount;
         } else {
-            $this->total_paid_real += Tools::ps_round(Tools::convertPrice($order_payment->amount, $order_payment->id_currency, false), 2);
+            $this->total_paid_real += Tools::ps_round(Tools::convertPrice($order_payment->amount, $this->id_currency, false), 2);
```

Comment:
I'm not sure this is correct Because `$order_payment->id_currency` is already initialized with `($currency ? $currency->id : $this->id_currency);`
I think in your case the fix should be at the moment this method is called, not changing this method behaviour as it could cause some BC breaks